app/src/main/java/cm/aptoide/pt/abtesting/AbTestCacheValidator.java

```diff
@@ -0,0 +1,35 @@
+package cm.aptoide.pt.abtesting;
+
+import java.util.HashMap;
+
+public class AbTestCacheValidator {
+
+  private HashMap<String, ExperimentModel> localCache;
+
+  public AbTestCacheValidator(HashMap<String, ExperimentModel> localCache) {
+    this.localCache = localCache;
+  }
+
+  public boolean validateCache(String id) {
+    return localCache.containsKey(id) && !localCache.get(id)
+        .hasError() && !localCache.get(id)
+        .getExperiment()
+        .isExperimentOver();
+  }
```

Comment:
this name doesn't look very clear to me, it looks like a void but it only validates cache, imo it should be something like isCacheValid